cmd/root.go

```diff
@@ -40,52 +32,43 @@ var (
 func init() {
 	rootCmd.Flags().StringVar(&cmdFlagAccount, cmdFlagKeyAccount, "", "Name of Bitrise account to use")
 	rootCmd.Flags().BoolVar(&cmdFlagPublic, cmdFlagKeyPublic, false, "Visibility of the Bitrise app")
-	rootCmd.Flags().StringVar(&cmdFlagRepo, cmdFlagKeyRepo, "", "Git URL for the repository to register")
-	rootCmd.Flags().StringVar(&cmdFlagStack, cmdFlagKeyStack, "", "The stack to run the builds on")
-	rootCmd.Flags().BoolVar(&cmdFlagAddWebhook, cmdFlagKeyAddWebhook, false, "To register a webhook for the git provider")
-	rootCmd.Flags().BoolVar(&cmdFlagAutoCodesign, cmdFlagKeyAutoCodesign, false, "Upload codesign files for iOS project")
 	rootCmd.Flags().StringVar(&cmdFlagAPIToken, cmdFlagKeyAPIToken, "", "Your Bitrise personal access token")
 }
 
 func executePhases(cmd cobra.Command, progress *phases.Progress) error {
 	if cmd.Flags().Changed(cmdFlagKeyAccount) {
-		progress.Account = &cmdFlagAccount
-	}
-	if progress.Account == nil {
+		progress.Account = cmdFlagAccount
+	} else {
 		account, err := phases.Account(cmdFlagAPIToken)
 		if err != nil {
 			return err
 		}
-		progress.Account = &account
+		progress.Account = account
 	}
 
 	if cmd.Flags().Changed(cmdFlagKeyPublic) {
-		progress.Public = &cmdFlagPublic
-	}
-	if progress.Public == nil {
+		progress.Public = cmdFlagPublic
+	} else {
 		public, err := phases.IsPublic()
 		if err != nil {
 			return err
 		}
-		progress.Public = &public
+		progress.Public = public
 	}
 
-	if cmd.Flags().Changed(cmdFlagKeyRepo) {
-		progress.Repo = &cmdFlagRepo
+	// repo
+	repoDetails, err := phases.Repo(progress.Public)
```

Comment:
I think this can be a flag also. I would only remove the flags which requires mixed logic to compose. Wdyt?

Reply:
the thing is that the tool needs to run in the directory of the project, so if the repo url is given, we need another set of info/actions still